stdlib/src/os/os.mojo

```diff
@@ -244,3 +244,53 @@ fn abort[
         print(message, flush=True)
 
     return abort[result]()
+
+
+# ===----------------------------------------------------------------------=== #
+# remove/unlink
+# ===----------------------------------------------------------------------=== #
+fn remove(path: String) raises:
+    """Removes the speficied file.
```

Comment:
**Suggestion** Be clear about the semantics about what happens if the argument passed is a path to a directory.  In Python, according to their [docs](https://docs.python.org/3/library/os.html#os.remove), if a directory is passed, an error is raised.

Pedantically, should we also spell out whether the path is relative or absolute?